### PR TITLE
Implementation of Accept redeem request service

### LINKED ISSUE #151: Accept redeem request service
- Implementation of AcceptRedeemRequestService
- Generate hashlock and secret.
- Create Message entry in messages table

### PR DESCRIPTION
**Pull request by abhayks1**
PR covers below:
- Implementation of accept redeem request service.
- RequestType check in accept stake request service.

Fixes #151

### CODE REVIEW

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (puneet-khushwani-eth) [inline src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: add this service to services.ts & repositories.ts

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    );
+  }
+
+  /**
+   * Fetches bounty amount from contract and returns bounty amount.
+   *
+   * @param redeemRequest Redeem request object.
+   * @return {string} Bounty amount.
+   */
+  private async getBountyAmount(redeemRequest: MessageTransferRequest): Promise<string> {
```
Comment (puneet-khushwani-eth) [inline src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: we have bounty in gateway table in facilitator db, should use from it

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    await this.messageTransferRequestRepository.save(redeemRequest);
+  }
+
+  /**
+   * It pre-calculates message hash.
+   * @param redeemRequest Redeem request object.
+   * @param hashLock Hash lock of acceptRedeem transaction.
+   * @return Returns message hash.
+   */
+  private calculateMessageHash(redeemRequest: MessageTransferRequest, hashLock: string): string {
```
Comment (puneet-khushwani-eth) [inline src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: this method seems to be an exact replica of one in accept stake request service. both should come from a util class may be. similar for `generateSecret`

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+    return messageHash;
+  }
+
+  /**
+   * Updates the message hash in messageTransferRequest repository after
+   * accepting redeem request. Accepting redeem requests adds a new entry
+   * into messages' repository with a message hash. That exact message
+   * hash is updated here in messageTransferRequest repository.
+   */
+  private async updateMessageHash(
```
Comment (puneet-khushwani-eth) [inline src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: documentation for params missing

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts (hunk 1 above)
Comment (abhayks1) [reply src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: Done!

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts (hunk 4 above)
Comment (abhayks1) [reply src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: Added

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts (hunk 3 above)
Comment (abhayks1) [reply src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: refactored to Utils.ts

File: src/services/redeem_and_unstake/AcceptRedeemRequestService.ts (hunk 2 above)
Comment (abhayks1) [reply src/services/redeem_and_unstake/AcceptRedeemRequestService.ts]: Good point.
However . Bounty can be changed in Gateway/CoGateway repository. It's always good to read the latest value from blockchain as final source of truth.
We will need bounty value from db while implementing bounty hunter. We can evaluate at that time whether we need bounty field or not.

File: src/Utils.ts [hunk 5]
```diff
@@ -94,6 +96,68 @@ const Utils = {
...
+   * It pre-calculates message hash.
+   *
+   * @param web3 Web3 instance.
+   * @param messageTransferRequest Redeem request object.
+   * @param hashLock Hash lock of acceptRedeem transaction.
+   * @param intentHash Stake/Redeem intent hash.
+   *
+   * @return Returns message hash.
+   */
+  calculateMessageHash(
```
Comment (puneet-khushwani-eth) [inline src/Utils.ts]: add test case please

Comment (abhayks1) [reply src/Utils.ts]: Added test for calculateMessageHash

**Pull request merged**